src/components/validation/validators/component_spec/sources.rs

```diff
@@ -71,6 +71,33 @@ impl Display for SourceMetrics {
     }
 }
 
+fn sum_counters(
+    metric_name: SourceMetrics,
+    metrics: &[&vector_core::event::Metric],
+) -> Result<f64, Vec<String>> {
+    let mut events: f64 = 0.0;
```

Comment:
To avoid the cast in the caller:

```suggestion
/// Sum the counters in the input metrics.
/// An error is appended to the result if any metric value is a non-counter.
/// `T` describes the numeric type of the metric values.
fn sum_counters<T>(
    metric_name: SourceMetrics,
    metrics: &[&vector_core::event::Metric],
) -> Result<T, Vec<String>> {
    let mut sum: T = 0.0;
```

Reply:
edit:

```Rust
fn sum_counters<T: Default>(
    metric_name: SourceMetrics,
    metrics: &[&vector_core::event::Metric],
) -> Result<T, Vec<String>> {
    let mut sum = T::default();
```

Reply:
I made a note about this above. I would still need some kind of cast since the metric type if a f64. I want to leave that type because we'll need it for certain metrics.